### PR TITLE
fix: Ability to increase file size limit

### LINKED ISSUE #2703: [FEATURE] Ability to increase file size limit
### Is your feature request related to a problem? Please describe.

I'm trying to create a survey, where people can upload images.
Some images will definitely be bigger than 10MB.

Right now I can't do that survey with Formbricks, because of the file size limit. 

### Describe the solution you'd like

Either
- Higher default size
- Ability to set default size with environment variable
- Ability to set higher limit than default in survey options
image

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by gupta-piyush19**
## What does this PR do?
- Ability to increase file size limit
    1. self hosted users can upload files of any size without any limit
- fixed partial files uploading logic

Fixes 
#2703 

## How should this be tested?

- Test A
- Test B

### Required

- [ ] Filled out the "How to test" section in this PR
- [ ] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [ ] Self-reviewed my own code
- [ ] Commented on my code in hard-to-understand bits
- [ ] Ran `pnpm build`
- [ ] Checked for warnings, there are none
- [ ] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [ ] My changes don't cause any responsiveness issues
- [ ] First PR at Formbricks? [Please sign the CLA!](https://cla-assistant.io/formbricks/formbricks) Without it we wont be able to merge it 🙏

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

File: packages/lib/utils/promises.ts [hunk 1]
```diff
@@ -1,3 +1,11 @@
 export const delay = (ms: number) => {
   return new Promise((resolve) => setTimeout(resolve, ms));
 };
+
+export function isFulfilled<T>(val: PromiseSettledResult<T>): val is PromiseFulfilledResult<T> {
```
Comment (Dhruwang) [inline packages/lib/utils/promises.ts:5]: We can convert this to arrow function 😊

File: packages/lib/utils/promises.ts [hunk 2]
```diff
@@ -1,3 +1,11 @@
...
+  return val.status === "fulfilled";
+}
+
+export function isRejected<T>(val: PromiseSettledResult<T>): val is PromiseRejectedResult {
```
Comment (Dhruwang) [inline packages/lib/utils/promises.ts:9]: We can convert this to arrow function 😊

File: apps/web/app/(app)/(survey-editor)/environments/[environmentId]/surveys/[surveyId]/edit/components/FileUploadQuestionForm.tsx [hunk 3]
```diff
@@ -193,7 +199,7 @@ export const FileUploadQuestionForm = ({
                 onChange={(e) => {
                   const parsedValue = parseInt(e.target.value, 10);
 
-                  if (parsedValue > maxSizeInMBLimit) {
+                  if (isFormbricksCloud && parsedValue > maxSizeInMBLimit) {
                     toast.error(`Max file size limit is ${maxSizeInMBLimit} MB`);
                     if (isFormbricksCloud) {
```
Comment (Dhruwang) [inline apps/web/app/(app)/(survey-editor)/environments/[environmentId]/surveys/[surveyId]/edit/components/FileUploadQuestionForm.tsx:204]: This check seems redundant, as we already check for `isFormbricksCloud` in if condition 😊

File: packages/lib/storage/service.ts [hunk 4]
```diff
@@ -211,8 +212,15 @@ export const getS3UploadSignedUrl = async (
   environmentId: string,
   isBiggerFileUploadAllowed: boolean = false
 ) => {
-  const maxSize = isBiggerFileUploadAllowed ? MAX_SIZES.big : MAX_SIZES.standard;
-  const postConditions: PresignedPostOptions["Conditions"] = [["content-length-range", 0, maxSize]];
+  const maxSize = IS_FORMBRICKS_CLOUD
```
Comment (Dhruwang) [inline packages/lib/storage/service.ts:215]: Not super important, but we can also create a function for calculating the max size in `utils.ts`. What do you think ? 🤔

File: packages/surveys/src/components/general/FileInput.tsx [hunk 5]
```diff
@@ -58,11 +59,22 @@ export const FileInput = ({
       const uploadPromises = filteredFiles.map((file) =>
         onFileUpload(file, { allowedFileExtensions, surveyId })
       );
-      const uploadedUrls = await Promise.all(uploadPromises);
+
+      const uploadedfiles = await Promise.allSettled(uploadPromises);
```
Comment (Dhruwang) [inline packages/surveys/src/components/general/FileInput.tsx:63]: Can we convert this to camelcase `uploadedfiles` -> `uploadedFiles` 😊

**Review by Dhruwang — CHANGES_REQUESTED** [review]
Hey Piyush, limit works nice 😊🚀
Made few comments for some minor code improvements

Not related to this PR, but when i tried to upload a larger file, uploading status does not get updated instantly, so it does not clearly indicate whether file is being uploaded or not 


https://github.com/formbricks/formbricks/assets/67850763/f6e66c17-6430-4f05-b545-744d2c5cfa43

File: packages/lib/storage/service.ts (hunk 4 above)
Comment (gupta-piyush19) [reply packages/lib/storage/service.ts:215]: I tried this, but constants are not available in utils, we might need to pass many arguments just to replace one ternary, we can look at this later.

**Review by Dhruwang — APPROVED** [review]
Good to go 🚀

**Pull request merged**